invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from "react"
+import { useListAllBoardsQuery } from "services/api/endpoints/boards";
+import { useListImagesQuery } from "services/api/endpoints/images";
+
+export const useHasImages = () => {
+    const { data: boardList, isLoading: loadingBoards } = useListAllBoardsQuery({ include_archived: true })
+    const { data: uncategorizedImages, isLoading: loadingImages } = useListImagesQuery({
+        board_id: "none",
+        offset: 0,
+        limit: 0,
+        is_intermediate: false,
+    })
+
+    const hasImages = useMemo(() => {
+        // default to true
+        if (loadingBoards || loadingImages) {
+            return true
+        }
```

Comment:
One improvement might be to return a sentinel value while loading and use that to display a loading indicator:

```ts
export const LOADING_SYMBOL = Symbol("LOADING")

export const useHasImages = () => {
    const { data: boardList, isLoading: loadingBoards } = useListAllBoardsQuery({ include_archived: true })
    const { data: uncategorizedImages, isLoading: loadingImages } = useListImagesQuery({
        board_id: "none",
        offset: 0,
        limit: 0,
        is_intermediate: false,
    })

    const hasImages = useMemo(() => {
        // default to true
        if (loadingBoards || loadingImages) {
            return LOADING_SYMBOL
        }

...
```

Then in `<NoContentForViewer />`:

```tsx
if (hasImages === LOADING_SYMBOL) {
   return <Loading />
}
```

Reply:
Nice, added